### PR TITLE
LB-331: Case insensitive messybrainz

### PR DESCRIPTION
**Pull request by paramsingh**
Lowercase everything before working with the database.

We should probably hold off on deploying this until the data we already have is fixed?

### CODE REVIEW

File: messybrainz/data.py [hunk 1]
```diff
@@ -241,3 +241,31 @@ def link_recording_to_recording_id(connection, msid, mbid):
...
+
+    def lower(d):
+        """ Lower cases all values in the specified dict.
+        """
+        result = {}
+        for key, value in d.items():
+            if isinstance(value, str):
+                result[key] = value.lower()
+            elif isinstace(value, dict):
+                result[key] = lower(value)
```
Comment (alastair) [inline messybrainz/data.py]: If you're doing these list/dict checks then it'd be good to extend the test to include these things too

File: messybrainz/data.py [hunk 2]
```diff
@@ -144,11 +144,11 @@ def submit_recording(connection, data):
     Returns:
         the Recording MessyBrainz ID of the data
     """
-    data_json = json.dumps(data, sort_keys=True, separators=(',', ':'))
+    data_json = _convert_to_messybrainz_json(data)
```
Comment (alastair) [inline messybrainz/data.py]: I wonder if we should only convert the data to lower-case while computing the sha.

In this example, `data_json` is now lower-case and we insert it into the database like that. This means that
1. it's different from what the user submitted
2. if we want to use it to show metadata on the website it's now all lower-case and ugly

as we get the artist name and release data from `data`, the artist and release data will be original case, but the data in the recording table will be lower-cased

**Review by mayhem — APPROVED** [review]
🔥 !!

**Comment by paramsingh** [comment]
Closing in favor of #56

**Pull request closed without merging**